XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -1797,9 +1797,12 @@ public bool CanAddVtpm(out string cannotReason)
         {
             cannotReason = null;
 
-            if (VTPMs.Count >= VM.MAX_ALLOWED_VTPMS)
+            if (Helpers.XapiEqualOrGreater_23_10_0(Connection))
+                return allowed_operations.Contains(vm_operations.create_vtpm);

```

Comment:
Here and in other pages we're not returning a `cannotReason`. I'd add message saying that the host doesn't support the operation.

Reply:
Yes, I did remark on the ticket that this change affects the UX. I was wondering about adding a message like this, but I was not sure whether it would add any value because it doesn't really explain why the host doesn't support it. I can add it though if you think that it's better for the user to see it.

Reply:
We can just add something like "creating vtpms is not a supported operation for this host" or something. It's better than nothing, I'd definitely want to see it